homeassistant/components/hausbus/device.py

```diff
@@ -0,0 +1,58 @@
+"""Representation of a Haus-Bus device."""
+
+from pyhausbus import HausBusDevice
+from pyhausbus.de.hausbus.homeassistant.proxy.controller.params.EFirmwareId import (
+    EFirmwareId,
+)
+
+from homeassistant.helpers.device_registry import DeviceInfo
+
+from .const import DOMAIN
+
+
+class HausbusDevice:
+    """Common base for Haus-Bus devices."""
+
+    def __init__(
+        self,
+        bridge_id: str,
+        device_id: str,
+        sw_version: str,
+        hw_version: str,
+        firmware_id: EFirmwareId,
+    ) -> None:
+        """Set up Haus-Bus device."""
+        self._device_id = device_id
+        self.manufacturer = "Haus-Bus.de"
+        self.model_id = "Controller"
+        self.name = f"Controller {self._device_id}"
+        self.software_version = sw_version
+        self.hardware_version = hw_version
+        self.bridge_id = bridge_id
+        self.firmware_id = firmware_id
+
+    @property
+    def device_id(self) -> str | None:
+        """Return a serial number for this device."""
+        return self._device_id
```

Comment:
Why do we type a function returning a `str` as `str | None`?
Also, the getter seems useless, I'd suggest to remove it.